### ISSUE TITLE
office-ui-fabric-react v7?

### ISSUE THREAD

**Issue by chadbr** [microsoft/angular-react#126]
Is there a timeline for upgrading this project to use office-ui-fabric-react v7?

We're starting a new add-in based on an Angular project - and we'd like to start with 'the latest'.

Should we start with the current release in hopes that v7 will come soon or?

Other alternatives we should be looking at?

Thanks, Chad

**Comment by aarongreenwald** [comment]
Hey, thanks for checking in. We also want to stay up-to-date, so we're in the same boat. We're looking at what it would take to upgrade now. We're also in middle of trying to upgrade our projects to Angular 8, so we're going to see if we can upgrade both dependencies together. 

Unfortunately I can't give a timeline right now, it'll depend on if we run into breaking changes that we need to overcome. 

What version of angular are you using? 

As far as alternatives: I don't know. There might be, but I'm not familiar with them. So I can't say :-)

**Comment by chadbr** [comment]
Hey Aaron,

We've recently upgraded to Angular 8.

As far as timeline - can you give a rough idea? Weeks? Months? Years?

I suppose for now we need to use the current version and try to get it working with A8.

As far as alternatives -- I'm just trying to understand if this is Microsoft's 'recommended solution' for using fabric components in Angular.

Thanks much, Chad

**Comment by aarongreenwald** [comment]
The timeline for Angular 8 is closer to weeks (please don't take that as a promise or commitment - just an educated guess from where we stand now). As we upgrade to Angular 8 we'll see what the impact on fabric is. 

If you do use the current version and get it working with A8, I'd very much appreciate if you can share whatever it is you learn back with the community. Might make this go faster. Pull requests are also welcome (but do check out the contributing guidelines). 

As far as Microsoft's 'recommended solution': I personally am not able to make a broad statement like that. 😀 You can ask over at the [fabric-core](https://github.com/OfficeDev/office-ui-fabric-core) and [fabric-react](https://github.com/OfficeDev/office-ui-fabric-react) repos if they have an official recommendation for Angular or if they're aware of other implementations. You can also always roll your own using fabric-core. We found it more convenient to use angular-react to wrap existing React components, but it's not your only option.

**Comment by aarongreenwald** [comment]
Ok, here's an update: the A8 upgrade is done (see #127). We decided not to try to upgrade the fabric version at the same time because the new fabric-react version is actual not just a version upgrade but a change in design systems, so it's a bigger decision that just upgrading versions. We'll probably be upgrading at some point in the future but we're not sure when and where it'll fit into our priorities. 

Just some context: when I say "we" here I'm referring to my team, a small team within Microsoft that maintains this project because we need it for the angular apps we develop and maintain. We're happy to share it as open source in the hopes that it can be useful to other people, but our primary focus is the products we are responsible for, and this is kind of a side-project for us. So for now the priority of the transition to fabric-react v7 is lower on our list, but that might change in the future. 

Bottom line: If you absolutely need fabric-react v7, I'd recommend looking for another solution or wrapping it yourself (you can use this project's `core` package and build your own angular-fabric on top of it, which might make sense). If you just want to use fabric-react and aren't really particular about which version, then this library might be a good fit for you.

**Comment by chadbr** [comment]
@aarongreenwald thanks for the update. For what it's worth - we seem to have successfully upgraded all the dependencies in the project and are using it with small changes.

I suppose we can just fork the repo and publish a new package with a different name (scoped)?

Or maybe we'll just fork it and publish internally.

Thanks for the update and the level setting. Appreciated.

**Comment by aarongreenwald** [comment]
> I suppose we can just fork the repo and publish a new package with a different name (scoped)?
> 
> Or maybe we'll just fork it and publish internally.

Either option, depending on your needs. I don't have a preference. :) Thanks for understanding!

**Comment by joseivan** [comment]
@chadbr do you plan to make your changes public? We are about to do same job you did so if you could share your work it'd be great. Thanks
